float32 x
float32 y
float32 theta
float32 phi
bool on
bool cartesian
bool velocity